src/libaktualizr/uptane/fetcher.cc

```diff
@@ -0,0 +1,114 @@
+#include "fetcher.h"
+
+#ifdef BUILD_OSTREE
+#include "package_manager/ostreemanager.h"  // TODO: Hide behind PackageManagerInterface
+#endif
+
+namespace Uptane {
+
+std::string Fetcher::fetchRole(const std::string& base_url, Uptane::Role role, Version version) {
+  // TODO: chain-loading root.json
+  std::string url = base_url + "/" + version.RoleFileName(role);
+  HttpResponse response = http.get(url);
+  if (!response.isOk()) {
+    throw MissingRepo(url);
+  }
+  return response.body;
+}
+
+bool Fetcher::fetchRoot(bool director, Version version) {
+  std::string root;
+  try {
+    root = fetchRole((director) ? config.uptane.director_server : config.uptane.repo_server, Role::Root(), version);
+  } catch (MissingRepo()) {
+    LOG_WARNING << "Can't connect to the repo";
+    return false;
+  }
+  storage->storeUncheckedRoot(director, root, version);
+  return true;
+}
+
+bool Fetcher::fetchMeta() {
```

Comment:
This paradigm is a little odd. It's legal to fetch just one or several things. This also won't really work when the root is `N > 1` versions behind what is on the server. It would need to fetch several roots before it could fetch the others. Or, if you do the update and have no new `timestamp` metadata, you can return early because that means the rest of the metadata is unchanged.

Reply:
Yes, I've postponed proper root fecthing until root versioning is implemented on the backend. `fetchRoot` and `feedRoot` are temporary hacks to keep uptane vector tests running.

Reply:
Ok, that's fair enough.